.github/workflows/main.yml

```diff
@@ -0,0 +1,50 @@
+name: Main workflow
+
+on:
+  push:
+    branches:
+      - master
```

Comment:
All steps should run on all pull requests. Only deploy should be on master branch

Reply:
main.yml is divided into two parts for CI and CD

Reply:
united workflows into one with the condition whether there is a push to the master